### PR TITLE
Feature/1447-write-end-to-end-tests-for-convert

### PR DESCRIPTION
**Pull request by martinbaerwolff**
OP#1447

### CODE REVIEW

**Comment by martinbaerwolff** [comment]
@randy-seng config tests fail, I assume because I removed 4 of the 8 video filetypes from the config dict. I tried to also remove them from the default_config.otvision.yaml, but got other errors. Could you have a look at what you could fix in the config or the yaml?

File: tests/convert/test_convert.py [hunk 1]
```diff
@@ -1,43 +1,89 @@
...
-            Path(h264_ref_video),
-            Path(h264_ref_video).parents[1]
-            / test_data_tmp_dir.name
-            / Path(h264_ref_video).name,
-        )
-    h264_test_videos = get_files(
-        paths=[test_data_tmp_dir],
-        filetypes=[".h264"],
+    # # Convert test h264 files
+    convert(
```
Comment (randy-seng) [inline tests/convert/test_convert.py:81]: Maybe use a local variable `test_case_dir` and `test_case_tmp_dir` and if used a lot throughout method?`

```python
test_case_convert_dir = test_convert_dir / test_case
test_case_convert_tmp_dir = test_convert_tmp_dir / test_case

def convert(
        paths=[test_case_convert_tmp_dir],
        output_filetype=output_filetype,
        input_fps=input_fps,
        fps_from_filename=fps_from_filename,
):
 # ...



```

File: tests/convert/test_convert.py [hunk 2]
```diff
@@ -61,29 +107,184 @@ def array_from_video(file: Path) -> np.ndarray:
...
+
+@pytest.mark.parametrize(
+    "output_filetype",
+    [".foo", ".jpg", ".mpeg", ".avchd", ".flv", ".swf", ".m4v", ".mpg", ".wmv"],
+)
+def test_fail_convert_not_supported_output_filetypes(
+    test_convert_tmp_dir: Path,
+    output_filetype: str,
+) -> None:
+    """Tests if the main function of OTVision/conver/conver.py raises specific
```
Comment (randy-seng) [inline tests/convert/test_convert.py:241]: Typo

```suggestion
    """Tests if the main function of OTVision/convert/convert.py raises specific
```

File: tests/convert/test_convert.py [hunk 3]
```diff
@@ -61,29 +107,184 @@ def array_from_video(file: Path) -> np.ndarray:
...
+)
+def test_fail_convert_wrong_parameters(
+    test_convert_tmp_dir: Path,
+    output_filetype: str,
+    input_fps: float,
+    fps_from_filename: bool,
+    overwrite: bool,
+    delete_input: bool,
+) -> None:
+    """Tests if the main function of OTVision/conver/conver.py raises specific
```
Comment (randy-seng) [inline tests/convert/test_convert.py:276]:
```suggestion
    """Tests if the main function of OTVision/convert/convert.py raises specific
```

**Review by randy-seng — CHANGES_REQUESTED** [review]
I have a few change requests~

File: tests/convert/test_convert.py (hunk 1 above)
Comment (martinbaerwolff) [reply tests/convert/test_convert.py:81]: Of course! :)

**Review by randy-seng — APPROVED** [review]
I pushed a quick fix of concerning the detect command line script. Just a wrong access to the CONFIG dict. So, good to merge!

**Review by randy-seng — APPROVED** [review]

**Review by randy-seng — APPROVED** [review]

**Pull request merged**